### PR TITLE
Add U50LV and U55C targets 

### PR DESCRIPTION
**Pull request by anilmartha**
@jornt-xilinx

I have added following targets and corresponding test cases. Could you please review and merge it?

U50LV
U50LV-DWC
U55C-DWC

### CODE REVIEW

File: python/pyxir/contrib/target/components/DPUCAHX8H/u50lv_dwc_op_support.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+        X=X,
+        bXs=bXs,
+        tXs=tXs,
+        channel_parallel=16,
+        bank_depth=2048,
+        max_pool_min_kernel=1,
+        max_pool_max_kernel=8,
+        max_pool_min_stride=1,
+        max_pool_max_stride=8,
+        avg_pool_min_kernel=2,
```
Comment (jornt-xilinx) [inline python/pyxir/contrib/target/components/DPUCAHX8H/u50lv_dwc_op_support.py:142]: avg_pool_min_kernel should be 1

File: python/pyxir/contrib/target/components/DPUCAHX8H/u50lv_op_support.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+        X=X,
+        bXs=bXs,
+        tXs=tXs,
+        channel_parallel=16,
+        bank_depth=2048,
+        max_pool_min_kernel=1,
+        max_pool_max_kernel=8,
+        max_pool_min_stride=1,
+        max_pool_max_stride=8,
+        avg_pool_min_kernel=2,
```
Comment (jornt-xilinx) [inline python/pyxir/contrib/target/components/DPUCAHX8H/u50lv_op_support.py:142]: avg_pool_min should be 1

File: python/pyxir/contrib/target/components/DPUCAHX8H/u50lv_op_support.py [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+
+@pyxir.register_op_support_check("DPUCAHX8H-u50lv", "ReLU6")
+def relu6_op_support(X: XLayer, bXs: List[XLayer], tXs: List[XLayer]) -> bool:
+    """Check whether we can execute the provided ReLU6 operator
+    on the DPUCAHX8H-u50lv target"""
+    assert len(bXs) == 1
+    bX = bXs[0]
+    return bX.type[0] in set(
+        ["Convolution", "Conv2DTranspose", "Eltwise", "BatchNorm", "Scale"]
```
Comment (jornt-xilinx) [inline python/pyxir/contrib/target/components/DPUCAHX8H/u50lv_op_support.py:211]: Relu6 is not supported after eltwise according to the doc. You can check with a unit test. We will also see this issue in the operator support of other DPUs.

File: python/pyxir/contrib/target/components/DPUCAHX8H/u50lv_dwc_op_support.py [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+
+
+@pyxir.register_op_support_check("DPUCAHX8H-u50lv_dwc", "ReLU6")
+def relu6_op_support(X: XLayer, bXs: List[XLayer], tXs: List[XLayer]) -> bool:
+    """Check whether we can execute the provided ReLU6 operator
+    on the DPUCAHX8H-u50lv_dwc target"""
+    assert len(bXs) == 1
+    bX = bXs[0]
+    return bX.type[0] in set(
+        ["Convolution", "Conv2DTranspose", "Eltwise", "BatchNorm", "Scale"]
```
Comment (jornt-xilinx) [inline python/pyxir/contrib/target/components/DPUCAHX8H/u50lv_dwc_op_support.py:211]: Relu6 is not supported after eltwise according to the doc. You can check with a unit test. We will also see this issue in the operator support of other DPUs.

File: python/pyxir/contrib/target/components/DPUCAHX8H/u55c_dwc_op_support.py [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+        X=X,
+        bXs=bXs,
+        tXs=tXs,
+        channel_parallel=16,
+        bank_depth=2048,
+        max_pool_min_kernel=1,
+        max_pool_max_kernel=8,
+        max_pool_min_stride=1,
+        max_pool_max_stride=8,
+        avg_pool_min_kernel=2,
```
Comment (jornt-xilinx) [inline python/pyxir/contrib/target/components/DPUCAHX8H/u55c_dwc_op_support.py:142]: avg_pool_min_kernel should be 1

File: python/pyxir/contrib/target/components/DPUCAHX8H/u55c_dwc_op_support.py [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+
+
+@pyxir.register_op_support_check("DPUCAHX8H-u55c_dwc", "ReLU")
+def relu_op_support(X: XLayer, bXs: List[XLayer], tXs: List[XLayer]) -> bool:
+    """Check whether we can execute the provided ReLU operator
+    on the DPUCAHX8H-u55c_dwc target"""
+    assert len(bXs) == 1
+    bX = bXs[0]
+    return bX.type[0] in set(
+        ["Convolution", "Conv2DTranspose", "Eltwise", "BatchNorm", "BiasAdd", "Scale"]
```
Comment (jornt-xilinx) [inline python/pyxir/contrib/target/components/DPUCAHX8H/u55c_dwc_op_support.py:200]: Relu6 is not supported after eltwise according to the doc. You can check with a unit test. We will also see this issue in the operator support of other DPUs.

File: python/pyxir/contrib/target/components/DPUCAHX8H/u55c_dwc_op_support.py [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+    return is_scale_supported(X, bXs, tXs, channel_parallel=16, bank_depth=2048)
+
+
+@pyxir.register_op_support_check("DPUCAHX8H-u55c_dwc", "Upsampling2D")
+def upsampling_op_support(X: XLayer, bXs: List[XLayer], tXs: List[XLayer]) -> bool:
+    """Check whether we can execute the provided Upsampling2D operator
+    on the DPUCAHX8H-u55c_dwc target"""
+    # TODO
+    method = X.attrs["method"]
+    return method == "nearest_neighbor"
```
Comment (jornt-xilinx) [inline python/pyxir/contrib/target/components/DPUCAHX8H/u55c_dwc_op_support.py:228]: We should check for:  `ceil(scale / bank_num) * scale * ceil(input_channel / channel_parallel) <= bank_depth`

File: python/pyxir/contrib/target/components/DPUCAHX8H/u50lv_op_support.py [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+    return is_scale_supported(X, bXs, tXs, channel_parallel=16, bank_depth=2048)
+
+
+@pyxir.register_op_support_check("DPUCAHX8H-u50lv", "Upsampling2D")
+def upsampling_op_support(X: XLayer, bXs: List[XLayer], tXs: List[XLayer]) -> bool:
+    """Check whether we can execute the provided Upsampling2D operator
+    on the DPUCAHX8H-u50lv target"""
+    # TODO
+    method = X.attrs["method"]
+    return method == "nearest_neighbor"
```
Comment (jornt-xilinx) [inline python/pyxir/contrib/target/components/DPUCAHX8H/u50lv_op_support.py:228]: We should check for:  `ceil(scale / bank_num) * scale * ceil(input_channel / channel_parallel) <= bank_depth`

File: python/pyxir/contrib/target/components/DPUCAHX8H/u50lv_dwc_op_support.py [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+    return is_scale_supported(X, bXs, tXs, channel_parallel=16, bank_depth=2048)
+
+
+@pyxir.register_op_support_check("DPUCAHX8H-u50lv_dwc", "Upsampling2D")
+def upsampling_op_support(X: XLayer, bXs: List[XLayer], tXs: List[XLayer]) -> bool:
+    """Check whether we can execute the provided Upsampling2D operator
+    on the DPUCAHX8H-u50lv_dwc target"""
+    # TODO
+    method = X.attrs["method"]
+    return method == "nearest_neighbor"
```
Comment (jornt-xilinx) [inline python/pyxir/contrib/target/components/DPUCAHX8H/u50lv_dwc_op_support.py:228]: We should check for:  `ceil(scale / bank_num) * scale * ceil(input_channel / channel_parallel) <= bank_depth`

File: python/pyxir/contrib/target/components/DPUCAHX8H/u55c_dwc_op_support.py [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+    """Check whether we can execute the provided Scale operator
+    on the DPUCAHX8H-u55c_dwc target"""
+    return is_scale_supported(X, bXs, tXs, channel_parallel=16, bank_depth=2048)
+
+
+@pyxir.register_op_support_check("DPUCAHX8H-u55c_dwc", "Upsampling2D")
+def upsampling_op_support(X: XLayer, bXs: List[XLayer], tXs: List[XLayer]) -> bool:
+    """Check whether we can execute the provided Upsampling2D operator
+    on the DPUCAHX8H-u55c_dwc target"""
+    return is_upscale_supported(X, bXs, tXs, channel_parallel=16, bank_depth=2048, bank_num=8)
```
Comment (jornt-xilinx) [inline python/pyxir/contrib/target/components/DPUCAHX8H/u55c_dwc_op_support.py:227]: Could you add some unit tests for upsampling?

File: python/pyxir/contrib/target/components/common/op_support.py [hunk 11]
```diff
@@ -297,3 +297,21 @@ def is_scale_supported(
...
+    tXs: List[XLayer],
+    channel_parallel: int,
+    bank_depth: int,
+    bank_num: int,
+) -> bool:
+    method = X.attrs["method"] 
+    input_channel = X.shapes[X.attrs["data_layout"].index("C")]
+    scale_h = X.attrs["scale_h"]
+    scale_w = X.attrs["scale_w"]
+    return (math.ceil(scale_h / bank_num) * scale_h * math.ceil(input_channel / channel_parallel) <= bank_depth) \
```
Comment (jornt-xilinx) [inline python/pyxir/contrib/target/components/common/op_support.py:314]: We can also support bilinear mode from the doc. Could you try adding that and corresponding tests?

Comment (anilmartha) [reply python/pyxir/contrib/target/components/common/op_support.py:314]: DPU supports only nearest_neighbor

[UNILOG][INFO] Compile mode: dpu
[UNILOG][INFO] Debug mode: function
[UNILOG][INFO] Target architecture: DPUCAHX8H_ISA2
[UNILOG][INFO] Graph name: quantize_eval_model, with op num: 11
[UNILOG][INFO] Begin to compile...
**[UNILOG][WARNING] xir::Op{name = upsample, type = upsample-fix} has been assigned to CPU: [DPU does not support BILINEAR mode. (only support NEAREST mode)]**

File: python/pyxir/contrib/target/components/DPUCAHX8H/u55c_dwc_op_support.py (hunk 10 above)
Comment (anilmartha) [reply python/pyxir/contrib/target/components/DPUCAHX8H/u55c_dwc_op_support.py:227]: Added

**Pull request merged**